python/examples/example_errors.ipynb

```diff
@@ -0,0 +1,977 @@
+    "\n",
+    "If any simulation failures occur, they will be printed via Python logging. \n",
+    "\n",
+    "Programmatically, simulation success can be checked via `ReturnDataView.status`. In case of a successful simulation, and only then, this value corresponds to `amici.AMICI_SUCCESS`.\n",
+    "In case of a simulation error, all quantities in `ReturnData`/`ReturnDataView` will be reported up to the time of failure, the rest will be `NaN`. The likelihood and it's gradient will always be `NaN` in case of failure."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0ad882ac",
+   "metadata": {},
+   "source": [
+    "## `AMICI_TOO_MUCH_WORK` - `mxstep steps taken before reaching tout`\n",
+    "\n",
+    "Let's run a simulation:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae3cb45e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "petab_problem = benchmark_models_petab.get_problem(\"Fujita_SciSignal2010\")\n",
+    "amici_model = import_petab_problem(petab_problem, verbose=False, force_compile=False)\n",
+    "\n",
+    "np.random.seed(2991)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e75a6dcc",
+   "metadata": {},
+   "source": [
+    "**What happened?**\n",
+    "\n",
+    "AMICI failed to integrate the forward problem. The problem occurred for only one simulation condition, `condition_step_03_0`. The issue occurred at $t = 3031.8$, where the CVODES reached the maximum number of steps.\n",
+    "\n",
+    "**How to address?**\n",
+    "\n",
+    "The number of steps the solver has to take is closely related to the chosen error tolerance. More accurate results, more steps. Therefore, this problem can be solved in two ways:\n",
+    "\n",
+    "1. Increasing the maximum number of steps via [`amici.Solver.setMaxSteps`](https://amici.readthedocs.io/en/latest/generated/amici.amici.Solver.html#amici.amici.Solver.setMaxSteps). Note that increasing the number of allowed steps increase RAM requirements, so don't set this to ridiculously large values in order to avoid this error.\n",
```

Comment:
Also note that setting high values opens the door to extremely long simulation times (that may still result in an error), which is probably the bigger issue in most practical situations.

Reply:
Agreed, runtime should me mentioned. 

I was referring more to changing that to something like 10**10, which will afaik require lots of RAM, even if fewer steps are taken.